examples/server/server.cpp

```diff
@@ -2572,7 +2572,7 @@ int main(int argc, char ** argv) {
 
     auto res_error = [](httplib::Response & res, json error_data) {
         json final_response {{"error", error_data}};
-        res.set_content(final_response.dump(), MIMETYPE_JSON);
+        res.set_content(final_response.dump(-1, ' ', false, json::error_handler_t::replace), MIMETYPE_JSON);
```

Comment:
> This patch adjusts the JSON dump call in the error handler to use the UTF-8 replacement character instead of throwing errors (matching the output dumping behavior that is used on successful output).

For successful output, we also use `.dump()` call without any arguments, so I don't get what do you mean by "matching the output dumping behavior that is used on successful output"

Example:

https://github.com/ggerganov/llama.cpp/blob/20f1789dfb4e535d64ba2f523c64929e7891f428/examples/server/server.cpp#L3214

Reply:
Completion and chat endpoints use replacement.

https://github.com/ggerganov/llama.cpp/blob/20f1789dfb4e535d64ba2f523c64929e7891f428/examples/server/server.cpp#L2974

Reply:
Either way, makes no sense to allow an error handler to crash the whole server over an encoding issue, since it causes the error itself not to get logged on caller side anymore.